app/services/proxy_status/pulp.rb

```diff
@@ -0,0 +1,46 @@
+require 'uri'
+
+module ProxyStatus
+  class Pulp < ::ProxyStatus::Base
+    def initialize(proxy, opts = {})
+      super(proxy, opts, 'Pulp')
+    end
+
+    def storage
+      fetch_proxy_data do
+        api.pulp_storage
+      end
+    end
+
+    def status
+      body = RestClient.get(pulp_url)
+      fail _("Pulp does not appear to be running.") if body.empty?
+      json = JSON.parse(body)
+      json['errors'] = {}
+
+      if json['known_workers'].empty?
+        json['errors']['known_workers'] = _("No pulp workers running.")
+      end
+
+      if json['database_connection'] && json['database_connection']['connected'] != true
+        json['errors']['database_connection'] = _("Pulp database connection issue.")
+      end
+
+      if json['messaging_connection'] && json['messaging_connection']['connected'] != true
+        json['errors']['messaging_connection'] = _("Pulp message bus connection issue.")
+      end
+
+      json
+    end
```

Comment:
I was actually thinking of doing the other way around (e.g., to create a separate PR to refactor https://github.com/Katello/katello/blob/master/app/models/katello/ping.rb#L105 to use this service class) :smile: 
Let me know which way you'd prefer.